### PR TITLE
MOEN-44169 logout completion callback

### PR DESCRIPTION
**Pull request by AnirudhaSM-engg-sdk**
### Jira Ticket
https://moengagetrial.atlassian.net/browse/MOEN-44169
### Description
added support for ios logout completion call back and base changes

### CODE REVIEW

File: sdk/core/src/moeParser/MoEngagePayloadParser.ts [hunk 1]
```diff
@@ -45,6 +46,20 @@ export function getUserDeletionData(payload: string): UserDeletionData {
...
 }
 
+/**
+ * Create an instance of {@link MoELogoutCompleteData} from json object
+ *
+ * @param payload - JSON Object with required keys
+ * @returns instance of {@link MoELogoutCompleteData}
+ * @since 12.7.0
+ */
+export function getLogoutCompleteData(payload: { [k: string]: any }): MoELogoutCompleteData {
```
Comment (msoumya-engg-sdk) [inline sdk/core/src/moeParser/MoEngagePayloadParser.ts]: **[Severity: Major]** This parser is not null-safe. If `payload` is `undefined`/`null`, or if `payload[ACCOUNT_META]` is missing/malformed, `getMoEAccountMeta` will throw and the event dispatch will fail silently inside `executeHandler`. Wrap in try/catch and return a nullable `MoELogoutCompleteData | null`, matching the Flutter `logoutCompleteDataFromJson` implementation which logs and returns `null` on invalid payloads.

File: sdk/core/src/models/MoELogoutCompleteData.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import MoEAccountMeta from "./MoEAccountMeta";
+import { MoEPlatform } from "./MoEPlatform";
+
+export default class MoELogoutCompleteData {
```
Comment (msoumya-engg-sdk) [inline sdk/core/src/models/MoELogoutCompleteData.ts:10]: **[Severity: Minor]** Missing JSDoc. Sibling model classes such as `UserDeletionData` include a class-level JSDoc (`@author`, `@since`) and per-field documentation. Add the same so this model matches the standard followed elsewhere in the SDK.

File: sdk/core/src/__mocks__/JsonDataProvider.ts [hunk 3]
```diff
@@ -27,4 +27,14 @@ export const expectedUserIdentityStringObjectTypePayload = {
...
+
+export const logoutCompleteIosPayload = JSON.stringify({
+    platform: "iOS",
+    accountMeta: { appId: appId }
+});
+
+export const logoutCompleteAndroidPayload = JSON.stringify({
+    platform: "android",
+    accountMeta: { appId: appId }
+});
```
Comment (msoumya-engg-sdk) [inline sdk/core/src/__mocks__/JsonDataProvider.ts]: **[Severity: Trivial]** Missing newline at end of file. Most existing source files in this repo end with a trailing newline; please add one here to match the convention.

File: sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts [hunk 4]
```diff
@@ -12,4 +14,20 @@ describe('MoEngagePayloadParser', () => {
...
+        });
+
+        it('Android payload should return MoELogoutCompleteData with android platform and correct appId', () => {
+            const result = getLogoutCompleteData(JSON.parse(logoutCompleteAndroidPayload));
+            expect(result).toBeInstanceOf(MoELogoutCompleteData);
+            expect(result.platform).toEqual(MoEPlatform.Android);
+            expect(result.accountMeta.appId).toEqual(appId);
+        });
+    });
 });
```
Comment (msoumya-engg-sdk) [inline sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts]: **[Severity: Trivial]** Missing newline at end of file. Please add a trailing newline to match the convention used elsewhere in the repo.

File: ExpoTVSampleApp/App.tsx [hunk 5]
```diff
@@ -71,6 +71,10 @@ function addMoEngageListeners() {
   ReactMoE.setEventListener("inAppCampaignSelfHandled", (payload) => {
     console.debug("inAppCampaignSelfHandled", payload);
   });
+
+  ReactMoE.setEventListener("logoutComplete", (data) =>
+    console.debug("logoutComplete", data)
+  );
```
Comment (msoumya-engg-sdk) [inline ExpoTVSampleApp/App.tsx:77]: **[Severity: Trivial]** Missing newline at end of file. Please add a trailing newline to match the convention used elsewhere in the repo.

File: sdk/core/src/utils/MoEEventHandlerHelper.ts [hunk 6]
```diff
@@ -25,6 +25,9 @@ export function executeHandler(
             } else if (type == PERMISSION_RESULT) {
                 let payload = getPermissionResult(notificationPayload)
                 handler(payload)
+            } else if (type == LOGOUT_COMPLETE) {
+                let logoutData = getLogoutCompleteData(notificationPayload);
+                handler(logoutData);
```
Comment (msoumya-engg-sdk) [inline sdk/core/src/utils/MoEEventHandlerHelper.ts]: **[Severity: Major]** No unit test covers this new branch in `executeHandler`. The dispatcher-level routing for `LOGOUT_COMPLETE` (parsing `notificationPayload`, invoking `handler` with the parsed `MoELogoutCompleteData`) is untested. Add a test in `MoEEventHandlerHelper.test.ts` (or equivalent) that mocks the handler and verifies it is called with the correctly parsed object. Also add an invalid-payload case to confirm no crash, matching the Flutter parser_test coverage for `logoutCompletePayloadInvalid`.

File: sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts [hunk 7]
```diff
@@ -12,4 +14,20 @@ describe('MoEngagePayloadParser', () => {
...
+            const result = getLogoutCompleteData(JSON.parse(logoutCompleteIosPayload));
+            expect(result).toBeInstanceOf(MoELogoutCompleteData);
+            expect(result.platform).toEqual(MoEPlatform.IOS);
+            expect(result.accountMeta.appId).toEqual(appId);
+        });
+
+        it('Android payload should return MoELogoutCompleteData with android platform and correct appId', () => {
+            const result = getLogoutCompleteData(JSON.parse(logoutCompleteAndroidPayload));
+            expect(result).toBeInstanceOf(MoELogoutCompleteData);
+            expect(result.platform).toEqual(MoEPlatform.Android);
```
Comment (msoumya-engg-sdk) [inline sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts]: **[Severity: Major]** Test coverage is incomplete for the new parser:

1. No invalid-payload test. The Flutter counterpart asserts `logoutCompleteDataFromJson(invalid)` returns `null`. Please add an equivalent case (e.g. payload missing `accountMeta`, or entirely malformed) so regressions in null-safety are caught.
2. Consider adding a test for an unknown/empty `platform` value to lock the expected behavior (throw, fallback, or ignore).

File: sdk/core/src/moeParser/MoEngagePayloadParser.ts (hunk 1 above)
Comment (AnirudhaSM-engg-sdk) [reply sdk/core/src/moeParser/MoEngagePayloadParser.ts]: Fixed. `getLogoutCompleteData` now returns `MoELogoutCompleteData | null` and wraps the construction in try/catch — if `payload[ACCOUNT_META]` is missing or malformed, it returns `null` instead of throwing.

File: sdk/core/src/models/MoELogoutCompleteData.ts (hunk 2 above)
Comment (AnirudhaSM-engg-sdk) [reply sdk/core/src/models/MoELogoutCompleteData.ts:10]: Fixed. Added class-level `@author` / `@since 12.7.0` and per-field `@since` docs, matching the pattern in `UserDeletionData`.

File: sdk/core/src/__mocks__/JsonDataProvider.ts (hunk 3 above)
Comment (AnirudhaSM-engg-sdk) [reply sdk/core/src/__mocks__/JsonDataProvider.ts]: Fixed. Added trailing newline.

File: sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts (hunk 4 above)
Comment (AnirudhaSM-engg-sdk) [reply sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts]: Fixed. Added trailing newline.

File: ExpoTVSampleApp/App.tsx (hunk 5 above)
Comment (AnirudhaSM-engg-sdk) [reply ExpoTVSampleApp/App.tsx:77]: Fixed. Added trailing newline.

File: sdk/core/src/utils/MoEEventHandlerHelper.ts (hunk 6 above)
Comment (AnirudhaSM-engg-sdk) [reply sdk/core/src/utils/MoEEventHandlerHelper.ts]: Fixed. Added `MoEEventHandlerHelper.test.ts` with two cases: (1) valid iOS payload — asserts handler is called once with a `MoELogoutCompleteData` instance carrying the correct platform and appId; (2) invalid payload missing `accountMeta` — asserts handler is not called. Also added the null guard (`logoutData != undefined ? handler(logoutData) : null`) to match the pattern used by other branches.

File: sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts (hunk 7 above)
Comment (AnirudhaSM-engg-sdk) [reply sdk/core/src/__tests__/moeParser/MoEngagePayloadParser.test.ts]: Fixed. Added an invalid-payload test (payload missing `accountMeta`) that asserts `getLogoutCompleteData` returns `null`. For unknown/empty platform: the current behavior lets the raw string pass through to the `MoEPlatform` field at runtime (TypeScript types are erased). This matches the Flutter parser which also does no platform validation — the value is passed as-is to the model. No test added for this case as the behavior is intentional.

**Review by msoumya-engg-sdk — APPROVED** [review]
LGTM
